### PR TITLE
additional ensemble samplers

### PR DESCRIPTION
**Pull request by istfer**
Currently two methods are implemented in the get.ensemble.samples function: halton and uniform.

Regarding #491, I just added additional methods to generate ensemble samples: sobol and torus from randtoolbox (we were already using the halton function from this package), and @mdietze's lhc function from emulator package.

@ashiklom is up for review

### CODE REVIEW

**Comment by ashiklom** [comment]
Looks good to me!

For what it's worth -- `if` statements are conceptually easier to understand, but one alternative for code that looks like this is to use the `switch` function. Here are [some examples](http://stackoverflow.com/questions/10393508/how-to-use-the-switch-statement-in-r-functions).

File: utils/R/ensemble.R [hunk 1]
```diff
@@ -62,8 +62,6 @@ read.ensemble.output <- function(ensemble.size, pecandir, outdir,
 ##' @return matrix of random samples from trait distributions
 ##' @export
 ##' @import randtoolbox
-##' @references Halton, J. (1964), Algorithm 247: Radical-inverse quasi-random point sequence, 
-##' ACM, p. 701, doi:10.1145/355588.365104.
 ##' @author David LeBauer
 get.ensemble.samples <- function(ensemble.size, pft.samples,env.samples,method="uniform") {
```
Comment (dlebauer) [inline utils/R/ensemble.R]: Is 'uniform' a sensible default?

File: utils/R/ensemble.R [hunk 2]
```diff
@@ -62,8 +62,6 @@ read.ensemble.output <- function(ensemble.size, pecandir, outdir,
 ##' @return matrix of random samples from trait distributions
```
Comment (dlebauer) [inline utils/R/ensemble.R]: remove 'random' from definition/ title. 

for method parameter list options preferably with guidance for choosing a method

**Comment by ashiklom** [comment]
:+1: to Dave's comment. More documentation is often better, and is a good idea here.

Also, I'm not sure how the sampler functions are implemented, but it might not be a bad idea to add `...` ([ellipsis](http://www.r-bloggers.com/r-three-dots-ellipsis/)) to the main definition and to all of the samplers that can take additional arguments. That will make this function more versatile.

**Comment by istfer** [comment]
when it was only halton and uniform @mdietze suggested we should go with uniform as the default, there was this issue #520  about halton for small ensemble sizes.

but I don't know what should it be now, sobol perhaps?

**Comment by mdietze** [comment]
Uniform is the default because random will be the most commonly used and is most consistent with expectations.

Don't know what's up with the random in your samplers.pdf, since the pattern there is highly periodic, which shouldn't be happening from a standard runif(0,1) and not at all what I get when I run that function by itself.

**Comment by istfer** [comment]
ups sorry my bad, wrong figure, nope it's not periodic
[uniform1000.pdf](https://github.com/PecanProject/pecan/files/206867/uniform1000.pdf)

**Comment by ashiklom** [comment]
Looks good to me!

**Pull request merged**